lint/lint.go

```diff
@@ -38,3 +39,24 @@ type Context struct {
 type Checker interface {
 	Check(f *ast.File) []Warning
 }
+
+var checkers = map[string]func(c *Context) Checker{
+	"param-name":        newParamNameChecker,
+	"type-guard":        newTypeGuardChecker,
+	"parenthesis":       newParenthesisChecker,
+	"underef":           newUnderefChecker,
+	"param-duplication": newParamDuplicationChecker,
+}
+
+func NewChecker(name string, ctx *Context) Checker {
+	return checkers[name](ctx)
```

Comment:
This will panic if there is no checker `name`.
But that's OK for now.

Reply:
How to properly handle this?

Reply:
In normal circumstance, I would return `(Checker, error)`.
Mostly because it's a bad practice to do panic or fatal (exit) on constructor-like function error.

```go
if c := checkers[name]; c != nil {
	return c(ctx), nil
}
return nil, fmt.Errorf("checker %s not found", name)
```
The caller may decide what to do: if it's main, it could do `log.Fatalf("can't create checker: %v", err)`,
it it's testing function, it will just call `t.Errorf(...)`, and so on.

It's also OK to panic here because function contract requires it to be called only with `AvailableCheckers` returned values. Do what you think is better. There is no single right decision here, it depends on how we look at it.

Reply:
Handling such error is usually useful in tests, but I'm not sure we're interested in testing this particular function. It's either called properly, or it's a bad programming mistake and panic will lead to error location (stack trace).

Reply:
Thank you for detailed answer!